esphome/components/hlw8032/hlw8032.cpp

```diff
@@ -0,0 +1,167 @@
+
+  if (((data == 0x55) || (data == 0xaa) || (data & 0xf0)) && !this->header_found_) {
+    this->header_found_ = true;
+    this->raw_data_[0] = data;
+  } else if (data == 0x5A && this->header_found_) {
+    this->raw_data_[1] = data;
+    this->raw_data_index_ = 2;
+    this->check_ = 0;
+  } else if (this->raw_data_index_ >= 2 && this->raw_data_index_ < 24) {
+    this->raw_data_[this->raw_data_index_] = data;
+    if (this->raw_data_index_ < 23) {
+      this->check_ += data;
+    }
+    this->raw_data_index_++;
+    if (this->raw_data_index_ == 24) {
+      if (this->check_ == this->raw_data_[23]) {
+        this->parse_data_();
+      } else
+        ESP_LOGW(TAG, "Invalid checksum from HLW8032: 0x%02X != 0x%02X", this->check_, this->raw_data_[23]);
+
+      this->raw_data_index_ = 0;
+      this->header_found_ = false;
+      memset(this->raw_data_, 0, 24);
+    }
+  }
+}
+
+void HLW8032Component::parse_data_() {
+  // Parse header
+  uint8_t state_reg = this->raw_data_[0];
+
+  if (state_reg == 0xAA) {
+    ESP_LOGE(TAG, "HLW8032's function of error correction fails.");
+    return;
+  }
+
+  // Parse data frame
+  uint32_t voltage_parameter = encode_uint24(this->raw_data_[2], this->raw_data_[3], this->raw_data_[4]);
+  uint32_t voltage_reg = encode_uint24(this->raw_data_[5], this->raw_data_[6], this->raw_data_[7]);
+  uint32_t current_parameter = encode_uint24(this->raw_data_[8], this->raw_data_[9], this->raw_data_[10]);
+  uint32_t current_reg = encode_uint24(this->raw_data_[11], this->raw_data_[12], this->raw_data_[13]);
+  uint32_t power_parameter = encode_uint24(this->raw_data_[14], this->raw_data_[15], this->raw_data_[16]);
+  uint32_t power_reg = encode_uint24(this->raw_data_[17], this->raw_data_[18], this->raw_data_[19]);
+
+  uint8_t data_update_register = this->raw_data_[20];
+
+  bool have_power = data_update_register & (1 << 4);
+  bool have_current = data_update_register & (1 << 5);
+  bool have_voltage = data_update_register & (1 << 6);
+
+  bool power_cycle_exceeds_range = false;
+  if ((state_reg & 0xF0) == 0xF0) {
+    if (state_reg & 0xF) {
+      ESP_LOGW(TAG, "HLW8032 reports: (0x%02X)", state_reg);
+      if (state_reg & (1 << 3)) {
+        ESP_LOGW(TAG, "  Voltage REG overflows.");
+        have_voltage = false;
+      }
+      if (state_reg & (1 << 2)) {
+        ESP_LOGW(TAG, "  Current REG overflows.");
+        have_current = false;
+      }
+      if (state_reg & (1 << 1)) {
+        ESP_LOGW(TAG, "  Power REG overflows.");
+        have_power = false;
+      }
+      if (state_reg & (1 << 0)) {
+        ESP_LOGW(TAG, "  Voltage Parameter REG, Current Parameter REG and Power Parameter REG is not usable.");
+        return;
+      }
+    }
+    power_cycle_exceeds_range = state_reg & (1 << 1);
+  }
+
+#if ESPHOME_LOG_LEVEL >= ESPHOME_LOG_LEVEL_VERY_VERBOSE
+  ESP_LOGVV(TAG, "HLW8032 Parsed data:");
+  ESP_LOGVV(TAG, "  Voltage Parameter REG: 0x%06X, Voltage REG: 0x%06X", (unsigned int) voltage_parameter,
```

Comment:
We don't need to log the component name as the TAG includes that already. Please check all log lines.
https://developers.esphome.io/architecture/logging/#minimizing-impact

Also, when printing 32-bit values, you need to use these macros. Please apply to the rest as well.
```suggestion
  ESP_LOGVV(TAG, "Parsed data:");
  ESP_LOGVV(TAG, "  Voltage Parameter REG: 0x%06" PRIX32 ", Voltage REG: 0x%06" PRIX32, voltage_parameter, voltage_reg);
```

Reply:
I have fixed this. Thanks.